modules/apps/document-library/document-library-web-test/src/testIntegration/java/com/liferay/document/library/web/internal/asset/util/test/DLFileEntryLayoutClassedModelUsageActionMenuContributorTest.java

```diff
@@ -0,0 +1,175 @@
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Gergely Szalay
+ */
+@FeatureFlags("LPD-36446")
+@RunWith(Arquillian.class)
+@Sync
+public class DLFileEntryLayoutClassedModelUsageActionMenuContributorTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE,
+			SynchronousDestinationTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_company = _companyLocalService.getCompany(
+			TestPropsValues.getCompanyId());
+
+		_group = GroupTestUtil.addGroup();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getGroupId());
+
+		_fileEntry = _addDLFileEntry(
+			RandomTestUtil.randomString(), RandomTestUtil.randomString());
+	}
+
+	@Test
+	public void testGetLayoutClassedModelUsageActionDropdownItems()
+		throws Exception {
+
+		LayoutClassedModelUsage layoutClassedModelUsage =
+			_addLayoutClassedModelUsage(_fileEntry);
+
+		HttpServletRequest mockHttpServletRequest = _getHttpServletRequest();
+
+		List<DropdownItem> dropdownItems =
+			_layoutClassedModelUsageActionMenuContributor.
+				getLayoutClassedModelUsageActionDropdownItems(
+					mockHttpServletRequest, layoutClassedModelUsage);
+
+		DropdownItem dropdownItem = dropdownItems.get(0);
+
+		String href = GetterUtil.getString(dropdownItem.get("href"));
+
+		Assert.assertNotNull(href);
+
+		Assert.assertTrue(
+			StringUtil.contains(
+				href, String.valueOf(layoutClassedModelUsage.getPlid()),
+				StringPool.BLANK));
+
+		Assert.assertEquals("View in Page", dropdownItem.get("label"));
+	}
+
+	private FileEntry _addDLFileEntry(String fileName, String content)
+		throws Exception {
+
+		return _dlAppLocalService.addFileEntry(
+			null, TestPropsValues.getUserId(), _group.getGroupId(),
+			DLFolderConstants.DEFAULT_PARENT_FOLDER_ID, fileName,
+			ContentTypes.TEXT_PLAIN, RandomTestUtil.randomString(),
+			StringPool.BLANK, StringPool.BLANK, StringPool.BLANK,
+			content.getBytes(), null, null, null, _serviceContext);
+	}
+
+	private LayoutClassedModelUsage _addLayoutClassedModelUsage(
+			FileEntry fileEntry)
+		throws Exception {
+
+		Layout layout = LayoutTestUtil.addTypePortletLayout(_group);
+
+		return LayoutClassedModelUsageLocalServiceUtil.
+			addLayoutClassedModelUsage(
+				_group.getGroupId(), PortalUtil.getClassNameId(FileEntry.class),
+				fileEntry.getFileEntryId(), StringPool.BLANK,
+				RandomTestUtil.randomString(), RandomTestUtil.randomLong(),
+				layout.getPlid(), _serviceContext);
```

Comment:
Can you use `@Inject` to get a reference to the service instead of using the static Util?